src/main/java/apprenticeship/Rules.java

```diff
@@ -0,0 +1,84 @@
+package apprenticeship;
+
+
+public class Rules {
+    private boolean hasWinner = false;
+    private boolean isADraw = false;
+    private String winner;
+    private boolean gameInProgress = true;
+
+    public boolean gameProgress(Board board) {
+        if (board.countEmptyCells() == 0) {
+            this.gameInProgress = false;
+        } else if (this.hasWinner) {
+            this.gameInProgress = false;
+        }
+        return this.gameInProgress;
+    }
+
+    public boolean checkForWinner(Board board) {
+        this.checkForRowWin(board);
+        this.checkForDiagonalWin(board);
+        this.checkForColumnWin(board);
+        return this.hasWinner;
+    }
+
+    public boolean endsInADraw() {
+        if (!hasWinner && !gameInProgress) {
+            this.isADraw = true;
+        }
+        return isADraw;
+    }
+
+    public String getWinner() {
+        return this.winner;
+    }
+
+    private boolean checkForRowWin(Board board) {
+        for (int cellIndex = 0; cellIndex <= 6; cellIndex += 3) {
```

Comment:
^ Yeah, it is a bit weird to see the board initialized with an arbitrary size, but then the rules only apply to a 3x3 board. Maybe stick with 3x3 for now?